src/joint_state_listener.cpp

```diff
@@ -143,10 +143,10 @@ void JointStateListener::callbackJointState(const JointStateConstPtr& state)
       joint_positions.insert(make_pair(state->name[i], state->position[i]));
     }
 
-    for (MimicMap::iterator i = mimic_.begin(); i != mimic_.end(); i++) {
-      if(joint_positions.find(i->second->joint_name) != joint_positions.end()) {
-        double pos = joint_positions[i->second->joint_name] * i->second->multiplier + i->second->offset;
-        joint_positions.insert(make_pair(i->first, pos));
+    for (auto & i : mimic_) {
```

Comment:
We can change this to use the new loop style, but again, let's not make it auto.  Also, we should have a more descriptive name for the variable than `i`.  Finally, can the loop iteration variable be `const` here?